test/performance/shmem_perf_suite/common.h

```diff
@@ -0,0 +1,83 @@
+/*
+*
+*  Copyright (c) 2015 Intel Corporation. All rights reserved.
+*  This software is available to you under the BSD license. For
+*  license information, see the LICENSE file in the top level directory.
+*
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <shmem.h>
+#include <shmemx.h>
+#include <string.h>
+#include <assert.h>
+
+char * aligned_buffer_alloc(int len) {
+    unsigned long page_align;
+    char *buf;
+
+    page_align = getpagesize();
+    buf = shmem_malloc(len + page_align);
+    buf = (char *) (((unsigned long) buf + (page_align - 1)) /
+            page_align * page_align);
+
+    return buf;
+}
+
+int static inline is_divisible_by_4(int num)
+{
+    assert(sizeof(int) == 4);
+    return (!(num & 0x00000003));
```

Comment:
I think this is unreliable when ```num``` is negative.  I think bitwise arithmetic on signed values is not portable/reliable since C does not define the representation.  Add ```assert(num >= 0)```?